clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp

```diff
@@ -673,51 +913,112 @@ inline sycl::event matmul(descriptor_ptr handle, matmul_desc_ptr compute_desc,
+        scalar_alpha_e = q_ptr->submit([&](sycl::handler &cgh) {
+          access_wrapper<const float *> alpha_acc(alpha, cgh);
+          sycl::accessor acc(buf, cgh, sycl::write_only, sycl::no_init);
+          cgh.single_task<dpct_kernel_name<class copy_alpha_dev_ptr>>(
+              [=]() { acc[0] = alpha_acc.get_raw_pointer()[0]; });
+        });
+      } else {
+        float alpha_host = *static_cast<const float *>(alpha);
+        scalar_alpha_e = q_ptr->submit([&](sycl::handler &cgh) {
+          sycl::accessor acc(buf, cgh, sycl::write_only, sycl::no_init);
+          cgh.single_task<dpct_kernel_name<class copy_alpha_host_ptr>>(
+              [=]() { acc[0] = alpha_host; });
+        });
+      }
+#else
+      scalar_alpha_e =
+          q_ptr->memcpy(scales_alpha->get_data_handle(), alpha, sizeof(float));
+#endif
     }
+    // alpha = alpha * scale_a * scale_b
+    sycl::event multiply_impl_e = detail::multiply_impl(
+        q_ptr, scales_alpha, compute_desc->_a_scale_pointer,
+        compute_desc->_b_scale_pointer,
+        std::vector<sycl::event>{scalar_alpha_e});
+    transform_events.push_back(multiply_impl_e);
+    transform_events.push_back(scalar_alpha_e);
     matmul_args.insert(
         {DNNL_ARG_ATTR_SCALES | DNNL_ARG_WEIGHTS, *scales_alpha});
   }
 
-  auto matmul_pd = ::dnnl::matmul::primitive_desc(
-      handle->get_engine(), src_md, weights_md, dst_md, matmul_attr);
+  if (compute_desc->_epilogue != epilogue_t::nop) {
+    ::dnnl::post_ops matmul_ops;
+    matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_relu, 0.f, 0.f);
+    matmul_attr.set_post_ops(matmul_ops);
+  }
+
+  auto matmul_pd =
+      beta_is_zero
+          ? ::dnnl::matmul::primitive_desc(handle->get_engine(), src_md,
+                                           weights_md, dst_md, matmul_attr)
+          : ::dnnl::matmul::primitive_desc(handle->get_engine(), src_md,
+                                           weights_md, bias_md, dst_md,
+                                           matmul_attr);
   auto matmul_prim = ::dnnl::matmul(matmul_pd);
   sycl::event matmul_prim_event = ::dnnl::sycl_interop::execute(
       matmul_prim, handle->get_engine_stream(), matmul_args, transform_events);
 
-  sycl::event scale_d_event;
+  // end of calling oneDNN
+
+  sycl::event scale_d_with_alpha_event;
   if (vector_alpha)
-    scale_d_event = detail::scale_d_with_vector_alpha(
+    scale_d_with_alpha_event = detail::scale_d_with_vector_alpha(
         q_ptr, m, n, new_d, d_type, alpha, scale_type, {matmul_prim_event});
-  // end of calling oneDNN
+
+  sycl::event amax_d_event;
+  if (auto amax_ptr = compute_desc->_amax_d_pointer) {
+    amax_d_event = detail::type_dispatch<detail::amax_impl>(
+        d_desc->_type, amax_ptr, new_d, new_ldd, d_desc->_rows, d_desc->_cols,
+        q_ptr,
+        std::vector<sycl::event>{matmul_prim_event, scale_d_with_alpha_event});
+    // TODO: Remove this wait. This wait is a WA to make sure the amax is
```

Comment:
need fix this todo.

Reply:
I removed the amax support now, and will support it later after runtime fix the bug.